services/provider/client/client.go

```diff
@@ -207,3 +207,20 @@ func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID stri
 
 	return cc.Client.ReportStatus(apiCtx, req)
 }
+
+func (cc *OCSProviderClient) PeerBlockPool(ctx context.Context, blockPoolName string, pool, token []byte) (*pb.PeerBlockPoolResponse, error) {
+	if cc.Client == nil || cc.clientConn == nil {
+		return nil, fmt.Errorf("OCS client is closed")
+	}
+
+	req := &pb.PeerBlockPoolRequest{
+		BlockPoolName: blockPoolName,
+		Pool:          pool,
+		Token:         token,
```

Comment:
should these values be encrypted on wire? these fields are providing less context, token is kinda ok, what should the pool be?

do you anticipate the params for this func increasing, if yes, can this be abstracted by interface like what we did here https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/interfaces/interfaces.go

Reply:
I will create a new PR for mTLS. Pool is the base64 encoded value for the blockPoolName, we can remove the blockPoolName and decode the pool when needed, WDYT?

No, I do think the params for PeerBlockPool will increase

Reply:
can be resolved.